src/derived/slot.rs

```diff
@@ -353,7 +354,7 @@ where
                             },
                         });
 
-                        let result = future.wait().unwrap_or_else(|| db.on_propagated_panic());
+                        let result = future.wait().unwrap_or_else(|| Canceled::throw());
```

Comment:
Is there no other sort of panic that can get propagated?

Reply:
I think we should’t propagate “normal” panics here. A panic will propagate to the original caller of the query anyway, so we don’t have to amplify it and can just cancel.

Roughly, if you have n parallel tasks which support cancellation, and one of the tasks dies, there’s no need to propagate the panic across tasks. It feels more natural to just orderly cancel the rest, and let only the original panic bubble up instead.

Reply:
OK, you convinced me, but I think this merits a comment.

Reply:
Added a comment